main.go

```diff
@@ -199,6 +209,14 @@ func main() {
 
 	// Start "Running" the backend network. This will block until the context is done so run in another goroutine.
 	go bn.Run(ctx)
+
+	// When running on GCE with cloud integration, the node comes up explicitly
+	// marked with NodeUnavailable status which flannel needs to reset after
+	// successfully provisioning a route on GCE
+	if config.BackendType == "gce" {
+		go resetNodeCondition(ctx)
```

Comment:
As far as I understood kubelet's code [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L244), the node status is reset on each kubelet start which means, the `NodeNetworkUnavailable` will be reset if kubelet is restarted.
My question hence - do we need to patch flannel to do this or this can be off-site? We (could) have some reconiliation loops as part of the (planet) agent which would enforce this as I don't see anything in particular that ties this functionality to flannel.

Reply:
No, this status is set only during initial node registration. It is not reset when kubelet restarts.